linux/mcompile/keymap/deadkey.cpp

```diff
@@ -0,0 +1,468 @@
+	v_dw_1D line;
+	line.push_back(convertNamesTo_DWORD_Value(first));
+	line.push_back(convertNamesTo_DWORD_Value(second));
+	//line.push_back(convertNamesTo_DWORD_Value(nameresult));
+	line.push_back(number);
+	return line;
+}
+
+std::vector<DeadKey*> create_alDead() {
+	std::vector<DeadKey*> alDead;
+	v_dw_2D dk_ComposeTable;
+
+  create_DKTable(dk_ComposeTable);
+
+	for( int i=0; i < (int) dk_ComposeTable.size()-1; i++) {
+		DeadKey *dk2 = new DeadKey(dk_ComposeTable[i][0]);
+		for ( int j=0; j< (int) dk_ComposeTable.size();j++) {
+			if(( dk_ComposeTable[i][0] == dk_ComposeTable[j][0]) && (IsKeymanUsedChar(dk_ComposeTable[j][1])))
+				dk2->KMX_AddDeadKeyRow(dk_ComposeTable[j][1],dk_ComposeTable[j][2]);
+		}
+		alDead.push_back(dk2);
+	}
+	return alDead;
+}
+
+void refine_alDead(KMX_WCHAR dk, std::vector<DeadKey*> &dkVec, std::vector<DeadKey*> *p_All_Vec) {
+	if( dk == 0)
+		return;
+
+	for (int j=0; j < (int) (*p_All_Vec).size()-1;j++) {
+		if( dk == (*p_All_Vec)[j]->KMX_GetDeadCharacter()) {
+			if(! found_dk_inVector(dk, dkVec)) {
+				dkVec.push_back((*p_All_Vec)[j]);
+				return;
+			}
+			else return;
+		}
+	}
+}
+
+bool found_dk_inVector(KMX_WCHAR dk, std::vector<DeadKey*> &dkVec) {
+	int i=0;
+	if( dkVec.size() > 0) {
+		do {
+			if( dk == dkVec[i]->KMX_GetDeadCharacter())
+				return true;
+			i++;
+		} while (i < (int) dkVec.size());
+	}
+	return false;
+}
+
+bool query_dk_combinations_for_specific_dk(v_dw_2D * p_dk_ComposeTable, v_dw_2D  &dk_SingleTable, KMX_DWORD dk) {
+	v_dw_1D line;
+
+	for ( int i =0; i< (int) (*p_dk_ComposeTable).size(); i++) {
+		if (((*p_dk_ComposeTable)[i][0] == dk) && (IsKeymanUsedChar((*p_dk_ComposeTable)[i][1]))) {
+			line.push_back((*p_dk_ComposeTable)[i][0]);
+			line.push_back((*p_dk_ComposeTable)[i][1]);
+			line.push_back((*p_dk_ComposeTable)[i][2]);
+			dk_SingleTable.push_back(line);
+			line.clear();
+		}
+	}
+
+	if( dk_SingleTable.size()>0)
+		return true;
+	else
+		return false;
+}
+
+KMX_DWORD KMX_changeKeynameToCapital(KMX_DWORD KVal, KMX_DWORD &shift, GdkKeymap* keymap) {
+  guint Keyval = (guint) KVal;
+  GdkKeymapKey* keys;
+  gint n_keys;
+
+	KMX_DWORD Cap = (KMX_DWORD) gdk_keyval_to_upper (KVal);
+	if( Keyval !=0) {
+		gdk_keymap_get_entries_for_keyval(keymap, Keyval, &keys, &n_keys);
+		for (int i = 0; i < n_keys; i++) {
+			if (keys[i].group == 0) {
+				shift = keys[i].level;
+				return Cap;
+			}
+		}
+	}
+	return Cap;
+}	
+
+void create_DKTable(v_dw_2D & dk_ComposeTable) {
+  //create a 2D-Vector which contains data for ALL existing deadkey combinations on a Linux Keyboard:
+  //dk_ComposeTable[i][0] : First    (e.g. dead_circumflex)
+  //dk_ComposeTable[i][1] : Second   (e.g. a)
+  //dk_ComposeTable[i][3] : Unicode-Value   (e.g. 0x00E2)
+
+  //values taken from: https://help.ubuntu.com/community/GtkDeadKeyTable#Accents
+	// _S2 Do we want to use GTK instead of this function?
+
+  v_dw_1D line;
```

Comment:
Why do you use a 1D vector here instead of creating a new struct, something like:

```
typedef struct {
  KMX_DWORD diacritic;
  KMX_DWORD base_char;
  KMX_DWORD unicode_value;
} DeadKeyMapping;
```

(There might be a better name for the struct...)

Reply:
(Should create an issue for this for a future refactoring/improvement)

Reply:
ok-see #11802